mipengine/controller/node_landscape_aggregator.py

```diff
@@ -345,79 +358,39 @@ def _gather_all_dataset_info(
+                    current_datasets[dataset_name] = [node_id]
+
             aggregated_datasets[data_model] = current_labels
             datasets_locations[data_model] = current_datasets
-    return datasets_locations, aggregated_datasets
-
-
-def _get_cdes_across_nodes(
-    nodes: Dict[str, NodeInfo],
-    datasets_per_node: Dict[str, Dict[str, Dict[str, str]]],
-) -> Dict[str, List[Tuple[str, CommonDataElements]]]:
-    nodes_cdes = {}
-    for node_id, datasets_per_data_model in datasets_per_node.items():
-        node_socket_addr = _get_node_socket_addr(nodes[node_id])
-        for data_model in datasets_per_data_model:
-            cdes = _get_node_cdes(node_socket_addr, data_model)
-            if data_model not in nodes_cdes:
-                nodes_cdes[data_model] = []
-            nodes_cdes[data_model].append((node_id, cdes))
-    return nodes_cdes
-
-
-def _get_datasets_locations_of_compatible_data_models(
-    compatible_data_models, datasets_locations
-):
-    return {
-        compatible_data_model: datasets_locations[compatible_data_model]
-        for compatible_data_model in compatible_data_models
-    }
 
+    datasets_locations_without_duplicates = {}
+    for data_model, dataset_locations in datasets_locations.items():
+        datasets_locations_without_duplicates[data_model] = {}
 
-def _get_compatible_data_models(
-    data_model_cdes_across_nodes: Dict[str, List[Tuple[str, CommonDataElements]]]
-) -> Dict[str, CommonDataElements]:
-    """
-    Each node has its own data models definition.
-    We need to check for each data model if the definitions across all nodes is the same.
-    If the data model is not the same across all nodes containing it, we log the incompatibility.
-    The data models with similar definitions across all nodes are returned.
-    Parameters
-    ----------
-        data_model_cdes_across_nodes: the data_models each node has
-    Returns
-    ----------
-        Dict[str, CommonDataElements]
-            the data models with similar definitions across all nodes
-    """
-    data_models = {}
-    for data_model, cdes_from_all_nodes in data_model_cdes_across_nodes.items():
-        first_node, first_cdes = cdes_from_all_nodes[0]
-        for node, cdes in cdes_from_all_nodes[1:]:
-            if not first_cdes == cdes:
-                logger.info(
-                    f"Node '{first_node}' and node '{node}' on data model '{data_model}' "
-                    f"have incompatibility on the following cdes: {first_cdes} and {cdes} "
-                )
-                break
-        else:
-            data_models[data_model] = first_cdes
-    return data_models
+        for dataset, node_ids in dataset_locations.items():
+            if len(node_ids) == 1:
+                datasets_locations_without_duplicates[data_model][dataset] = node_ids[0]
+            else:
+                del aggregated_datasets[data_model][dataset]
+                _log_duplicated_dataset(node_ids, data_model, dataset)
```

Comment:
I think this is problematic. Will it work if 3 nodes have the same dataset name? Please also add a unit test for this.